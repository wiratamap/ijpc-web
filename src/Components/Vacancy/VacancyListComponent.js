import React from 'react';
import PropTypes from 'prop-types';

import VacancyComponent from './VacancyComponent';

const VacancyListComponent = ({ vacancies }) => (
  vacancies.content.map((vacancy) => (
    <VacancyComponent key={vacancy.id} vacancy={vacancy} />
  ))
);

VacancyListComponent.propTypes = {
  vacancies: PropTypes.shape({
    content: PropTypes.arrayOf(PropTypes.any)
  })
};

VacancyListComponent.defaultProps = {
  vacancies: {
    content: []
  }
};

export default VacancyListComponent;
