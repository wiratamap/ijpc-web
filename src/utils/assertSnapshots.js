import React from 'react';
import renderer from 'react-test-renderer';

/**
 * @param Component - (Component) Component that needs to be test.
 * @param testConfiguration - (object) object of props and test description
 */
const assertSnapshot = (Component, testConfiguration) => {
  const { props, description } = testConfiguration;

  describe('#render', () => {
    it(description, () => {
      const component = renderer.create(<Component {...props} />).toJSON();

      expect(component).toMatchSnapshot();
    });
  });
};

const assertSnapshots = (Component, testConfigurations) => {
  testConfigurations.forEach((testConfiguration) => assertSnapshot(Component, testConfiguration));
};

export default assertSnapshots;
