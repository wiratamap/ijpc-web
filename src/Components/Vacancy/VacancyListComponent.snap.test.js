import assertSnapshots from '../../utils/assertSnapshots';
import VacancyListComponent from './VacancyListComponent';
import VACANCIES from '../../fixtures/vacanciesData';

jest.mock('../../utils/numberFormat', () => ({
  currencyFormat: jest.fn().mockImplementation(() => 'IDR 0.00')
}))
  .mock('moment', () => () => ({
    fromNow: jest.fn().mockImplementation(() => '1 days ago'),
    format: jest.fn().mockImplementation(() => '1 Jan')
  }));

describe('VacancyListComponent snapshot', () => {
  const configurations = [
    {
      description: 'should render vacancy list component',
      props: {
        vacancies: VACANCIES
      }
    }
  ];
  assertSnapshots(VacancyListComponent, configurations);
});
