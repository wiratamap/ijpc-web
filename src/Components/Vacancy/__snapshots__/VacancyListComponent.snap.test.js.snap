// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VacancyListComponent snapshot #render should render vacancy list component 1`] = `
Array [
  <div
    className="card"
  >
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
        <div
          className="row m-0"
        >
          <div
            className="col-2"
          >
            <img
              alt="1"
              className="img-thumbnail image-thumbnail-item"
              src="placeholder-image-768x576.png"
            />
          </div>
          <div
            className="col-6"
          >
            <h4>
              VP of Engineering
            </h4>
            <p
              className="font-weight-light"
            >
              My Company Inc
            </p>
            <p
              className="font-weight-light text-small"
            >
              West Jakarta, Indonesia
               
              路
               
              IDR 0.00
               
              -
               
              IDR 0.00
               
              /
               
              month
            </p>
          </div>
          <div
            className="col-4"
          >
            <div
              className="row justify-content-end"
            >
              <p
                className="font-weight-light text-small"
              >
                Posted
                 
                1 days ago
                 
                路 Apply before
                 
                1 Jan
                .
              </p>
            </div>
            <div
              className="row justify-content-end"
            >
              <button
                className="btn btn-outline-primary"
                type="button"
              >
                Vacancy Detail
              </button>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>,
  <div
    className="card"
  >
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
        <div
          className="row m-0"
        >
          <div
            className="col-2"
          >
            <img
              alt="1"
              className="img-thumbnail image-thumbnail-item"
              src="placeholder-image-768x576.png"
            />
          </div>
          <div
            className="col-6"
          >
            <h4>
              Credit Analyst
            </h4>
            <p
              className="font-weight-light"
            >
              Your Company Inc
            </p>
            <p
              className="font-weight-light text-small"
            >
              West Jakarta, Indonesia
               
              路
               
              IDR 0.00
               
              -
               
              IDR 0.00
               
              /
               
              month
            </p>
          </div>
          <div
            className="col-4"
          >
            <div
              className="row justify-content-end"
            >
              <p
                className="font-weight-light text-small"
              >
                Posted
                 
                1 days ago
                 
                路 Apply before
                 
                1 Jan
                .
              </p>
            </div>
            <div
              className="row justify-content-end"
            >
              <button
                className="btn btn-outline-primary"
                type="button"
              >
                Vacancy Detail
              </button>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>,
]
`;
